wrappers/wrapper.rb

```diff
@@ -1608,16 +1618,22 @@ def shift_route_times(route, shift_amount, shift_start_index = 0)
       raise 'Cannot shift the route, there are not enough activities' if shift_start_index > route[:activities].size
 
       route[:start_time] += shift_amount if shift_start_index == 0
+      current_shift = shift_amount
       route[:activities].each_with_index{ |activity, index|
         next if index < shift_start_index
 
         activity[:begin_time] += shift_amount
         activity[:end_time] += shift_amount if activity[:end_time]
-        activity[:waiting_time] -= [shift_amount, activity[:waiting_time]].min if activity[:waiting_time]
-        activity[:departure_time] += shift_amount if activity[:departure_time]
+
+        if activity[:waiting_time]
+          waiting_resorb = [activity[:waiting_time] - current_shift, 0].max
+          current_shift = [current_shift - waiting_resorb, 0].max
+          activity[:waiting_time] -= waiting_resorb
+        end
+        activity[:departure_time] += current_shift if activity[:departure_time]
```

Comment:
The idea is that if waiting_time > 0. Then this waiting time will absorb as much as possible this shift so the next steps won't have to be shifted of the entire shift.

It is not really related to the change of travel_time == 0 to point_id == point_id, but as I changed the waiting time computation else where I had to fix it here too

Reply:
My confusion was because of my assumptions, on the other PR, I realized that you are using this function in other places so there can be waiting_time.

However, I still don't understand why we apply the `current_shift` "logic" (i.e., "absorb" corrected shift) only to `departure_time` but not also to `end_time`

Also, I think we should correct the `[:start]` with `current_shift` as well, no?

Reply:
:point_up:  this might be a bug :thinking:

Reply:
Yes ! I have fixed this logic in #369 but didn't transpose it here. Nice catch !